### PR TITLE
PJR-136 - add modularized minimal ePub reader based on the sample app included in the Readium V2 Kotlin Toolkit

### PR DESCRIPTION
**Pull request by roywatson**
This commit also includes: PJR-151, PJR-152, PJR-153, PJR-154.
[PJR-151: Add base Reader Fragment(s)](https://jira.nypl.org/browse/PJR-151)
[PJR-152: Create an Activity to host the Reader Fragment(s)](https://jira.nypl.org/browse/PJR-152)
[PJR-153: Create an ePub Specific Reader Fragment](https://jira.nypl.org/browse/PJR-153)
[PJR-154: Create ViewModel/Repository](https://jira.nypl.org/browse/PJR-154)
[PJR-136: Modularize Components in Readium Kotlin toolkit V2](https://jira.nypl.org/browse/PJR-136)

This implements the basic requirements for Milestone 1 and includes the above ticket and PJR-136.
This is the code that represents the demonstration given on 23 Jan 2023.

### CODE REVIEW

File: apps/oeh/ios/App/Podfile [hunk 1]
```diff
@@ -15,7 +15,6 @@ def capacitor_pods
   pod 'CapacitorFilesystem', :path => '../../../../node_modules/@capacitor/filesystem'
   pod 'CapacitorPreferences', :path => '../../node_modules/@capacitor/preferences'
   pod 'CapacitorSplashScreen', :path => '../../../../node_modules/@capacitor/splash-screen'
-  pod 'CordovaPlugins', :path => '../capacitor-cordova-ios-plugins'
```
Comment (samanthaandrews) [inline apps/oeh/ios/App/Podfile:18]: @ettore just checking that we are ok to remove this

Comment (roywatson) [reply apps/oeh/ios/App/Podfile:18]: Looking back I think that must have happened in the "npm run sync:static" step.

Comment (ettore) [reply apps/oeh/ios/App/Podfile:18]: actually I thought I already removed it in the PR I merged yesterday

Comment (roywatson) [reply apps/oeh/ios/App/Podfile:18]: I pushed the branch this PR is based on very early Saturday. It was so early I messed up creqting the PR and then only re-created it yesterday.So I guess I sort of inadvertently beat you to it ;-)

**Review by ettore — COMMENTED** [review]
all looks good to me here, although I can't comment on Android specific stuff.

the one comment I have is about the checked-in epub: do we need a third epub checked in? If there's something specific about this one, I would rather add it to the common `apps/oeh/ePubs` directory we already have instead of creating a new directory inside  `apps/oeh/android`, so that we have one place to look and every platform can benefit. 

On the other hand if this new ePub does _not_ add anything specific, I would not check this in and just use one of the 2 epubs we already have.

File: apps/oeh/android/app/src/main/java/org/openebooks/app/plugins/readerplugin/ui/EpubReaderFragment.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if(model == null) {
+            Log.v("RHAW", "&&&&&&&&&&&&&&&&&& model is null &&&&&&&&&&&&&&&&&&&&&")
+        }
+        publication = model.publication
+
+        childFragmentManager.fragmentFactory =
+            EpubNavigatorFragment.createFactory(
+                publication = publication,
+                baseUrl = null,
+                initialLocator = model.currLocator,
```
[the remark this answers is not in the document]
Comment (roywatson) [reply apps/oeh/android/app/src/main/java/org/openebooks/app/plugins/readerplugin/ui/EpubReaderFragment.kt:38]: It's not persisted at all yet, pending discussion of hybid vs. native responsibility for persistance so for now it's just scafolding awaiting a persitance decision.

File: apps/oeh/android/app/src/main/java/org/openebooks/app/plugins/readerplugin/ui/ReaderActivity.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                                    R.id.activity_container,
+                                    it,
+                                    Bundle(),
+                                    READER_FRAGMENT_TAG
+                                )
+                            }
+                        }
+                    }
+                    is ReaderViewModel.OpenPubResult.Error -> {
+                        Log.v("RHAW", "ReaderActivity::onStart() Error ${result.message}")
```
[the remark this answers is not in the document]
Comment (roywatson) [reply apps/oeh/android/app/src/main/java/org/openebooks/app/plugins/readerplugin/ui/ReaderActivity.kt:65]: Yes for now logging to the Android Studio LogCat (android specific). 

Agreed that we need to develop a universal logging strategy.

**Comment by roywatson** [comment]
> On the other hand if this new ePub does _not_ add anything specific, I would not check this in and just use one of the 2 epubs we already have.

The ebook has to be in a specific location. Is there a way to have the build CI/CD copy one of the other ePubs to that project folder before running the build?

**Comment by ettore** [comment]
> > On the other hand if this new ePub does _not_ add anything specific, I would not check this in and just use one of the 2 epubs we already have.
> 
> The ebook has to be in a specific location. Is there a way to have the build CI/CD copy one of the other ePubs to that project folder before running the build?

as we discussed on slack, it's probably not worth the effort. I assumed there was an easy way to do so from within the Android project; Xcode takes "ownership" of (or controls) everything in the project, so I assumed too much. BTW to be clear, the level of ownership that Xcode takes has its own drawbacks  :)

**Review by ettore — APPROVED** [review]

**Review by MalcolmMcFly — APPROVED** [review]

**Comment by MalcolmMcFly** [comment]
@roywatson @kristojorg  and I have removed the Vercel branch protections. You should now be in the green to merge.

**Pull request merged**